covid_model/model_specs.py

```diff
@@ -117,34 +118,37 @@ def build(cls, specs=None, engine=None,
         return specs
 
     def write_to_db(self, engine, schema='covid_model', table='specifications', tags=None):
-        metadata = MetaData(schema=schema)
-        metadata.reflect(engine, only=['specifications'])
-        specs_table = metadata.tables[f'{schema}.{table}']
+        specs_table = get_sqa_table(engine, schema=schema, table=table)
 
         if tags is not None:
             self.tags.update(tags)
 
-        stmt = specs_table.insert().values(
-            created_at=dt.datetime.now(),
-            base_spec_id=int(self.base_spec_id),
-            tags=self.tags,
-            start_date=self.start_date,
-            end_date=self.end_date,
-            tslices=self.tslices,
-            tc=self.tc,
-            tc_cov=self.tc_cov,
-            model_params=self.model_params,
-            vacc_actual={dose: rates.unstack(level='age').to_dict(orient='list') for dose, rates in self.actual_vacc_df.to_dict(orient='series').items()},
-            vacc_proj_params=self.vacc_proj_params,
-            vacc_proj={dose: rates.unstack(level='age').to_dict(orient='list') for dose, rates in self.proj_vacc_df.to_dict(orient='series').items()} if self.proj_vacc_df is not None else None,
-            timeseries_effects=self.timeseries_effects,
-            attribute_multipliers=self.attribute_multipliers
-        )
-
-        conn = engine.connect()
-        result = conn.execute(stmt)
-
-        self.spec_id = result.inserted_primary_key[0]
+        with Session(engine) as session:
+            max_spec_id = session.query(func.max(specs_table.c.spec_id)).scalar()
+            self.spec_id = max_spec_id + 1
```

Comment:
Was this previously an autoincrement in postgres? I can guess that it'd be best to do manually rather than rely on correct configuration of the database. Was this the rationale?